From 5f8f3b1227d11b0027c5789a97d639c5e697eecf Mon Sep 17 00:00:00 2001
From: Sebastian Niciarz <sebastian.niciarz@intel.com>
Date: Wed, 26 Aug 2015 17:38:08 +0200
Subject: [PATCH] fdp_nfc: Integrate FDP rel.
 fieldspeak_bx_gmin_android_l_3.2.8.0

Android maintenance release of Fields Peak B2
HAL 3.2.8.0 / OTP 6 / RAM 4.3
FDP rel. tag: fieldspeak_bx_gmin_android_l_3.2.8.0

Change log:

git log --pretty=format:"%h %ai | %s%d [%an]" ec713f..fieldspeak_bx_gmin_android_l_3.2.8.0
ad753e46 2015-07-30 15:29:31 +0200 | fix kw issue : Pointer 'p_device' checked for NULL in function fdp_remove [Zhang, Lei R]
60e88c4d 2015-07-09 09:54:38 +0200 | modify fdp resume and suspend to uncoditionally invoke fdp_clk_req() [Alexey Chekmarev]

Change-Id: If65c296d4c9b0ce00f80c91b4e99f65451e21757
Tracked-On: https://jira01.devtools.intel.com/browse/AKP-2414
Signed-off-by: Sebastian Niciarz <sebastian.niciarz@intel.com>
---
 drivers/staging/fdp/fdp_custom.c | 18 ++++++++++++++----
 1 file changed, 14 insertions(+), 4 deletions(-)

diff --git a/drivers/staging/fdp/fdp_custom.c b/drivers/staging/fdp/fdp_custom.c
index d403a8b8..241952b 100644
--- a/drivers/staging/fdp/fdp_custom.c
+++ b/drivers/staging/fdp/fdp_custom.c
@@ -837,8 +837,10 @@ int fdp_suspend(struct device *dev)
 		return 0;
 	}
 
-	/* if (p_device->state == CUSTOM_OPENED)
-		disable_irq(p_device->irqout); */
+/*
+	if (p_device->state == CUSTOM_OPENED)
+		disable_irq(p_device->irqout);
+*/
 
 	fdp_clk_req(p_device, REF_CLOCK_RELEASE);
 
@@ -856,8 +858,10 @@ int fdp_resume(struct device *dev)
 		return 0;
 	}
 
-	/* if (p_device->state == CUSTOM_OPENED)
-		enable_irq(p_device->irqout); */
+/*
+	if (p_device->state == CUSTOM_OPENED)
+		enable_irq(p_device->irqout);
+*/
 
 	fdp_clk_req(p_device, REF_CLOCK_RELEASE);
 
@@ -877,6 +881,12 @@ static int fdp_remove(struct i2c_client *client)
 
 	p_device = fdp_p_device;
 
+	if (!p_device) {
+		pr_err
+			("fdp_remove: p_device is missing\n");
+		return -ENODEV;
+	}
+
 	/* disable to wake the device using this IRQ */
 	irq_set_irq_wake(p_device->irqout, 0);
 
-- 
1.9.1

